math-delimiters.el

```diff
@@ -129,14 +220,17 @@ delimeters."
                                        (- (point) (length close))))
                 (flip-after (from-open from-close to-open to-close)
                             (delete-char (- (length from-close)))
+                            (math-delimiters--slurp-or-barf-characters from-close)
                             (insert to-close)
                             (let ((end (point)))
                               (backward-char (length to-close))
                               (search-backward from-open)
                               (delete-char (length from-open))
                               (insert to-open)
                               (goto-char (+ end (- (length to-open)
-                                                   (length from-open))))))
+                                                   (length from-open)))))
+                            (unless math-delimiters-compressed-display-math
+                              (math-delimiters--toggle-line-breaks to-open to-close)))
```

Comment:
When using the uncompressed display math option, we are first doing the compressed case anyways and then post-process things.  We could also differentiate a bit sooner, but this would require proper extra functions for the compressed and uncompressed version.  This has the advantage that `math-delimiters--slurp-or-barf-characters` is _almost_ compeltely unaware of this distinction and it results in (marginally) prettier code.

I don't think I notice any speed difference between these two options.  I have kept a (hopefully working) version of the other approach around; in case you strongly prefer that approach I can try to replace the current implementation with that